x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/indexlifecycle/LockableLifecycleType.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+package org.elasticsearch.xpack.core.indexlifecycle;
+
+import org.elasticsearch.common.io.stream.StreamOutput;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+
+public class LockableLifecycleType implements LifecycleType {
+    public static final String TYPE = "lockable";
+    public static final LockableLifecycleType INSTANCE = new LockableLifecycleType();
+
+    @Override
+    public List<Phase> getOrderedPhases(Map<String, Phase> phases) {
+        return new ArrayList<>(phases.values());
+    }
+
+    @Override
+    public List<LifecycleAction> getOrderedActions(Phase phase) {
+        return new ArrayList<>(phase.getActions().values());
+    }
+
+    @Override
+    public void validate(Collection<Phase> phases) {
+    }
+
+    @Override
+    public String getWriteableName() {
+        return TYPE;
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) {
+
+    }
```

Comment:
I wondered that, but it is never needed it seems.

Since the class itself has the logic to what to do with the method calls, there are no useful instance variables to serialize across